programs/survival/renewable_elytra_trade.sc

```diff
@@ -0,0 +1,65 @@
+// This app will make leatherworkers get a trade to sell elytra.
+// The conditions for the trade to appear and the trade tiself are fully customizable.
+// The trade appears once the conditions are met and you interact with the villager.
+
+// Custommize conditions
+
+global_in_the_end = false; // does the villager need to be in the end for the trade to appear?
+global_min_level = 1; // minimum level for the trade to appear
+global_hold_elytra = true; // does the player need to hold an elytra while interacting with the leatherworker to make the trade appear?
+global_consume_elytra = true; // is the elytra consumed when right clicking the villager with it to add the trade?
+global_chances = 100; // percent chance for the trade to appear when all other conditions are met
+global_consume_if_failed = false; // should the elyta be consumed even if adding the trade failed due to random chance (set in global_chances)
+
+
+// Customize the trade
+
+global_trade = encode_nbt(
+	{
+		'sell' -> {'id' -> 'minecraft:elytra', 'Count'->1},
+		'buy' -> {'id' -> 'minecraft:emerald', 'Count' -> 40},
+		'buyB' -> {'id' -> 'minecraft:phantom_membrane', 'Count' -> 5},
+		'uses' -> 0,
+		'demand' -> -20,
+		'priceMutiplier' -> 0.0,
+		'specialPrice' -> 0,
+		'rewardExp' -> 1, // boolean (gives/doesn't give xp)
```

Comment:
Maybe you could add this as config option, so like have a `global_should_give_xp = true` or smth, and change this line to `'rewardExp' -> number(global_should_give_xp )`?

Reply:
Hm, I'm not sure I like it. the configs decide how the appearing of the trade behaves. The trade editing is separate and should only be messed with if you have a minimal understanding of vanilla commands/how trades work. I don't want to mix the trade characteristics with the conditions for it to appear.

Reply:
Except for price, all values there are default for a custom trade, but could be modified for a more interesting experience, if you know what you are doing.

Reply:
I think it's a nice decision split appearing condition and the trade.